### PR TITLE
Add OT Trace Propagator.

### LINKED ISSUE #1391: Clarify the name used for Lightstep's (ot-tracer-*) propagator
Currently in some repos this propagator is called `OpenTracingPropogator` in others is called `OtPropagator`. Couple suggestions:
* Use consistent names, probably `LightstepPropagator`, since this is not the official OpenTracing propagator (or maybe I am wrong, but if that is the case then we need to probably add it to the ot-shim).

### PR DESCRIPTION
**Pull request by carlosalberto**
Fixes #1391 

Adds  `OT Trace` propagator as a (very) optional extension component.

cc @yurishkuro

### CODE REVIEW

**Review by mwear — APPROVED** [review]
LGTM

File: specification/context/api-propagators.md [hunk 1]
```diff
@@ -330,6 +330,13 @@ organization and MUST be distributed as OpenTelemetry extension packages:
 * [B3](https://github.com/openzipkin/b3-propagation).
 * [Jaeger](https://www.jaegertracing.io/docs/latest/client-libraries/#propagation-format).
 
+This a list of additional propagators that MAY be maintained and distributed
```
Comment (ocelotl) [inline specification/context/api-propagators.md:333]:
```suggestion
This is a list of additional propagators that MAY be maintained and distributed
```

File: specification/context/api-propagators.md [hunk 2]
```diff
@@ -330,6 +330,13 @@ organization and MUST be distributed as OpenTelemetry extension packages:
...
+as OpenTelemetry extension packages:
+
+* [OT Trace](https://github.com/opentracing?q=basic&type=&language=). Propagation format
+  used by the OpenTracing Basic Tracers. It MUST not use `OpenTracing` in the resulting
+  propagator name as it is not wide adopted format in the OpenTracing ecosystem.
```
Comment (ocelotl) [inline specification/context/api-propagators.md:338]:
```suggestion
  propagator name as it is not widely adopted format in the OpenTracing ecosystem.
```

**Review by ocelotl — CHANGES_REQUESTED** [review]

File: specification/context/api-propagators.md [hunk 3]
```diff
@@ -330,6 +330,13 @@ organization and MUST be distributed as OpenTelemetry extension packages:
 * [B3](https://github.com/openzipkin/b3-propagation).
 * [Jaeger](https://www.jaegertracing.io/docs/latest/client-libraries/#propagation-format).
 
+This is a list of additional propagators that MAY be maintained and distributed
+as OpenTelemetry extension packages:
+
+* [OT Trace](https://github.com/opentracing?q=basic&type=&language=). Propagation format
+  used by the OpenTracing Basic Tracers. It MUST not use `OpenTracing` in the resulting
```
Comment (yurishkuro) [inline specification/context/api-propagators.md:337]:
```suggestion
  used by the OpenTracing Basic Tracers. It MUST NOT use `OpenTracing` in the resulting
```

https://tools.ietf.org/html/rfc2119

**Review by yurishkuro — APPROVED** [review]

**Review by ocelotl — APPROVED** [review]

**Review by codeboten — APPROVED** [review]

**Review by bogdandrutu — APPROVED** [review]

**Pull request merged**